From a8543d9a9a95f4ab4912c987e60a9c26b4a8dc77 Mon Sep 17 00:00:00 2001
From: Sriram R <quic_srirrama@quicinc.com>
Date: Wed, 10 May 2023 18:01:40 +0530
Subject: [PATCH] ath12k: pass center freq and center freq2 to FW for EHT320
 mode

When vdev start is sent to FW for bringup in 320MHz BW, the center
freq1 and fre2 are not passed properly. The center freq1 is the
center freq of the primary 160MHz and center freq2 is the center
frequency of the whole 320MHz BW. Update this info in the channel
arg before passing to FW in vdev start/restart command

Signed-off-by: Sriram R <quic_srirrama@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index a9be0f0..a1ef9f2 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -1107,7 +1107,15 @@ static void ath12k_wmi_put_wmi_channel(struct wmi_channel *chan,

	chan->mhz = chan_arg.freq;
	chan->band_center_freq1 = chan_arg.band_center_freq1;
-	if ((chan_arg.mode == MODE_11AX_HE160) || (chan_arg.mode == MODE_11BE_EHT160)) {
+	if (chan_arg.mode == MODE_11BE_EHT320) {
+		if (chan_arg.freq > chan_arg.band_center_freq1)
+			chan->band_center_freq1 = center_freq1 + 80;
+		else
+			chan->band_center_freq1 = center_freq1 - 80;
+
+		chan->band_center_freq2 = chan_arg.band_center_freq1;
+
+	} else if ((chan_arg.mode == MODE_11AX_HE160) || (chan_arg.mode == MODE_11BE_EHT160)) {
		if (chan_arg.freq > chan_arg.band_center_freq1)
			chan->band_center_freq1 = center_freq1 + 40;
		else
-- 
2.17.1

